Framework/DataObjects/src/LeanPeak.cpp

```
@@ -0,0 +1,314 @@
+ * DetectorID, row and column will be set to -1 since they are not (necessarily)
+ *found.
+ * You can call findDetector to look for the detector ID
+ *
+ * @param qLab :: Q of the center of the peak, in reciprocal space.
+ *        This is in inelastic convention: momentum transfer of the LATTICE!
+ *        Also, q does have a 2pi factor = it is equal to 2pi/wavelength (in
+ *        Angstroms).
+ */
+void LeanPeak::setQLabFrame(const Mantid::Kernel::V3D &qLab,
+                            boost::optional<double>) {
+  this->setQSampleFrame(getInverseGoniometerMatrix() * qLab);
+}
+
+/** Set sample position
+ *
+ */
+void LeanPeak::setSamplePos(double, double, double) {
+  throw Exception::NotImplementedError("LeanPeak has no sample information");
+}
+
+/** Set sample position
+ *
+ */
+void LeanPeak::setSamplePos(const Mantid::Kernel::V3D &) {
+  throw Exception::NotImplementedError("LeanPeak has no sample information");
+}
+
+// -------------------------------------------------------------------------------------
+/** Return the detector position vector */
+Mantid::Kernel::V3D LeanPeak::getDetPos() const {
+  throw Exception::NotImplementedError("LeanPeak has no detector information");
+}
+
+// -------------------------------------------------------------------------------------
+/** Return the sample position vector */
+Mantid::Kernel::V3D LeanPeak::getSamplePos() const {
+  throw Exception::NotImplementedError("LeanPeak has no sample information");
+}
+
+// -------------------------------------------------------------------------------------
+/** Return the L1 flight path length (source to sample), in meters. */
+double LeanPeak::getL1() const {
+  return std::numeric_limits<double>::quiet_NaN();
+}
+
+// -------------------------------------------------------------------------------------
+/** Return the L2 flight path length (sample to detector), in meters. */
+double LeanPeak::getL2() const {
+  return std::numeric_limits<double>::quiet_NaN();
```

Should throw an exception